ntbs-service-unit-tests/Services/AlertServiceTest.cs

```diff
@@ -176,6 +170,143 @@ public async Task DismissAllOpenAlerts_DismissesOpenAlerts()
             _mockAlertRepository.Verify(r => r.SaveAlertChangesAsync(NotificationAuditType.Edited), Times.Once);
         }
 
+        [Fact]
+        public async Task DismissingDuplicateNotificationAlert_AlsoRemovesAlertOnDuplicateRecord()
+        {
+            // Arrange
+            var notification = new Notification
+            {
+                NotificationId = 1,
+                NotificationStatus = NotificationStatus.Notified
+            };
+            
+            var duplicateNotification = new Notification
+            {
+                NotificationId = 2,
+                NotificationStatus = NotificationStatus.Notified
+            };
+
+            var alert = new DataQualityPotentialDuplicateAlert
+            {
+                AlertId = 101,
+                AlertStatus = AlertStatus.Open,
+                DuplicateId = duplicateNotification.NotificationId
+            };
+            
+            var duplicateAlert = new DataQualityPotentialDuplicateAlert
+            {
+                AlertId = 102,
+                AlertStatus = AlertStatus.Open,
+                DuplicateId = notification.NotificationId
+            };
+            
+            _mockAlertRepository
+                .Setup(s => s.GetAllOpenAlertsByNotificationId(notification.NotificationId))
+                .Returns(Task.FromResult(new List<Alert>{alert}));
+            
+            _mockAlertRepository
+                .Setup(s => s.GetAllOpenAlertsByNotificationId(duplicateNotification.NotificationId))
+                .Returns(Task.FromResult(new List<Alert>{duplicateAlert}));
+
+            // Act
+            await _alertService.DismissAllOpenAlertsForNotification(notification.NotificationId);
+            
+            // Assert
+            AssertAlertClosedRecently(alert);
+            AssertAlertClosedRecently(duplicateAlert);
+        }
+        
+        [Fact]
+        public async Task DismissingDuplicateNotificationAlert_OnlyRemovesRelatedNotification_OnDuplicateRecords()
+        {
+            // This test looks at the edge case where there are three duplicate notifications
+            // pointing to one another in a triangle. Closing a notification at one vertex should close the duplicate 
+            // alerts along its connecting paths, but the duplicate alerts between the other two vertices
+            // should remain.
+            
+            // Arrange
+            var notification1 = new Notification
+            {
+                NotificationId = 1, NotificationStatus = NotificationStatus.Notified
+            };
+            
+            var notification2 = new Notification
+            {
+                NotificationId = 2, NotificationStatus = NotificationStatus.Notified
+            };
+
+            var notification3 = new Notification
+            {
+                NotificationId = 3, NotificationStatus = NotificationStatus.Notified
+            };

```

Comment:
I don't think you need all this notification setup as the alerts service method doesn't look at the notification itself. I could be wrong?

Reply:
You are right, and the same is true of the previous test also! I've changed these now :)